packages/server/lib/plugins/child/run_plugins.js

```diff
@@ -125,6 +127,7 @@ class RunPlugins {
     })
     .then((modifiedCfg) => {
       debug('plugins file successfully loaded')
+      modifiedCfg && validateNonMigratedOptions(modifiedCfg)
```

Comment:
Seems like we should be able to leverage the existing config validation functions here: https://github.com/cypress-io/cypress/blob/10.0-release/packages/config/lib/index.js#L119